qiskit_experiments/randomized_benchmarking/rb_experiment.py

```diff
@@ -162,3 +164,37 @@ def _generate_circuit(
                 rb_circ.measure_all()
                 circuits.append(rb_circ)
         return circuits
+
+    def run(
+        self,
+        backend: Backend,
+        analysis: bool = True,
+        experiment_data: Optional[ExperimentData] = None,
+        **run_options,
+    ) -> ExperimentData:
+        """Run an experiment and perform analysis.
+
+        Args:
+            backend: The backend to run the experiment on.
+            analysis: If True run analysis on the experiment data.
+            experiment_data: Optional, add results to existing
+                experiment data. If None a new ExperimentData object will be
+                returned.
+            run_options: backend runtime options used for circuit execution.
+
+        Returns:
+            The experiment data object.
+        """
+        if not self.analysis_options.error_dict:
```

Comment:
I don't think you need to override run, this should be able to be done in the analysis class `_run_analysis` method using something like:
```python
def _run_analysis(..., **options):
    error_dict = options['error_dict']
    qubits = experiment_data.metadata()['physical_qubits']
    if not error_dict:
        error_dict = error_dict = RBUtils.get_error_dict_from_backend(
             experiment_data.backend, qubits)
```

Reply:
Done